internal/controller/biosversion_controller.go

```diff
@@ -0,0 +1,907 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package controller
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/go-logr/logr"
+	"github.com/ironcore-dev/controller-utils/clientutils"
+	"github.com/ironcore-dev/controller-utils/conditionutils"
+	metalv1alpha1 "github.com/ironcore-dev/metal-operator/api/v1alpha1"
+	"github.com/ironcore-dev/metal-operator/bmc"
+	"github.com/ironcore-dev/metal-operator/internal/bmcutils"
+	"github.com/stmcginnis/gofish/common"
+	"github.com/stmcginnis/gofish/redfish"
+)
+
+// BIOSVersionReconciler reconciles a BIOSVersion object
+type BIOSVersionReconciler struct {
+	client.Client
+	ManagerNamespace string
+	Insecure         bool
+	Scheme           *runtime.Scheme
+	BMCOptions       bmc.BMCOptions
+}
+
+const (
+	biosVersionFinalizer                   = "firmware.ironcore.dev/biosversion"
+	biosVersionUpgradeIssued               = "biosversionUpgradeIssued"
+	biosVersionUpgradeCompleted            = "biosVersionUpgradeCompleted"
+	biosVersionUpgradeRebootServerPowerOn  = "biosVersionUpgradePowerOn"
+	biosVersionUpgradeRebootServerPowerOff = "biosVersionUpgradePowerOff"
+	biosVersionUpgradeVerficationCondition = "biosVersionUpgradeVerification"
```

Comment:
Because @afritzler complained about casing, we might want change it here as well.